File: app/src/main/java/home/self/beerviewer_mvvm/view/splash/SplashActivity.java
package home.self.beerviewer_mvvm.view.splash;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.Toast;

import com.bumptech.glide.Glide;

import javax.inject.Inject;

import dagger.android.support.DaggerAppCompatActivity;
import home.self.beerviewer_mvvm.R;
import home.self.beerviewer_mvvm.databinding.ActivitySplashBinding;
import home.self.beerviewer_mvvm.view.beersview.BeersViewActivity;

public class SplashActivity extends DaggerAppCompatActivity implements SplashNavigator {

    @Inject
    SplashViewModel splashViewModel;

    private ActivitySplashBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_splash);
        binding.setViewModel(splashViewModel);

        splashViewModel.takeView(this);
    }

    @Override
    protected void onResume() {
        super.onResume();
        splashViewModel.subscribe();
    }

    @Override
    protected void onPause() {
        super.onPause();
        splashViewModel.unsubscribe();
    }

    @Override
    public void showFailureMessage(String msg) {
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show();
    }

    @Override
    public void showSplashAnimation() {
        Glide.with(this)
                .asGif()
                .load(R.raw.beer_splash)
                .into(binding.beerAnimation);
    }

    @Override
    public void startBeerViewActivity() {
        Intent intent = new Intent(SplashActivity.this, BeersViewActivity.class);
        startActivity(intent);
        finish();
    }
}

File: app-kotlin/src/main/java/home/self/beerviewer_mvvm/app_kotlin/view/splash/SplashActivity.kt
package home.self.beerviewer_mvvm.app_kotlin.view.splash

import android.os.Bundle
import com.bumptech.glide.Glide
import home.self.beerviewer_mvvm.app_kotlin.base.BaseActivity
import home.self.beerviewer_mvvm.app_kotlin.base.Constants
import home.self.beerviewer_mvvm.app_kotlin.R
import home.self.beerviewer_mvvm.app_kotlin.view.beersview.BeersViewActivity
import kotlinx.android.synthetic.main.activity_splash.*
import org.jetbrains.anko.startActivity
import org.jetbrains.anko.toast

internal class SplashActivity : BaseActivity<SplashViewModel.ViewModel>() {

    override fun getLayoutRes(): Int = R.layout.activity_splash

    override fun getViewModel(): Class<SplashViewModel.ViewModel> = SplashViewModel.ViewModel::class.java

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        subscribe()

        showSplashAnimation()
    }

    private fun subscribe() {
        viewModel.outputs.errorMessage().observe { message ->
            toast(message)
        }

        viewModel.outputs.isSplashLoadingEnd().observe { pair ->
            startActivity<BeersViewActivity>(
                    Constants.EXTRA_DEFAULT_PAGE to pair
            ).apply { finish() }
        }
    }

    private fun showSplashAnimation() {
        Glide.with(this)
                .asGif()
                .load(R.raw.beer_splash)
                .into(beer_animation)
    }

}
